### PR TITLE
[MI-2669]: Created a hook for intercepting redux changes and moved the central login into it, formatted code

### PR DESCRIPTION
**Pull request by avas27JTG**
- Created a hook for intercepting redux changes and moved the central login into it
- Formatted code

### CODE REVIEW

**Review by raghavaggarwal2308 — APPROVED** [review]

File: webapp/src/hooks/usePluginReduxSyncState.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+import usePluginApi from 'hooks/usePluginApi';
+import useApiRequestCompletionState from 'hooks/useApiRequestCompletionState';
+
+/**
+ * This hook intercepts the redux changes for the plugin to perform common actions like
+ * checking if the user's account is connected
+ * closing any open modal if user is logged out from other client tab
+ * preventing opening a modal if user's account is not connected
+ * fetching linked project list whenever needed like RHS is opened, new project is linked, etc
```
Comment (ayusht2810) [inline webapp/src/hooks/usePluginReduxSyncState.ts:20]:
```suggestion
 * This hook intercepts the redux changes for the plugin to perform common actions like
 * checking if the user's account is connected
 * closing any open modal if the user is logged out from another client tab
 * preventing opening a modal if the user's account is not connected
 * fetching linked project list on the opening of RHS, on linking of a new project, etc
```

**Review by ayusht2810 — APPROVED** [review]

File: webapp/src/hooks/usePluginReduxSyncState.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                dispatch(toggleShowTaskModal({isVisible: true, commandArgs}));
+            }
+        }
+    }, [modalId, isConnected]);
+
+    useEffect(() => {
+        if (!isConnected) {
+            dispatch(toggleShowLinkModal({isVisible: false, commandArgs}));
+            dispatch(toggleShowSubscribeModal({isVisible: false, commandArgs}));
+            dispatch(toggleShowTaskModal({isVisible: false, commandArgs}));
```
Comment (manojmalik20) [inline webapp/src/hooks/usePluginReduxSyncState.ts:70]: Should we keep this in the above useEffect?

File: webapp/src/hooks/usePluginReduxSyncState.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import useApiRequestCompletionState from 'hooks/useApiRequestCompletionState';
+
+/**
+ * This hook intercepts the redux changes for the plugin to perform common actions like
+ * checking if the user's account is connected
+ * closing any open modal if the user is logged out from another client tab
+ * preventing opening a modal if the user's account is not connected
+ * fetching linked project list on the opening of RHS, on linking of a new project, etc
+ */
+function usePluginReduxSyncState() {
```
Comment (manojmalik20) [inline webapp/src/hooks/usePluginReduxSyncState.ts:22]:
```suggestion
function usePluginReduxState() {
```
I don't understand the meaning of `sync` here.

Comment (avas27JTG) [reply webapp/src/hooks/usePluginReduxSyncState.ts:22]: actually here we are reading some redux states and based on that updating/modifying other global redux states of the plugin like modals and fetching the list of project list, etc
So, in a way, we are syncing the plugin's redux states internally, also this name was suggested by mickmister [here](https://github.com/mattermost/mattermost-plugin-azure-devops/pull/141#discussion_r1081985432)

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**